cdap-spark-core-base/src/main/scala/co/cask/cdap/app/runtime/spark/SerializableSparkExecutionContext.scala

```diff
@@ -128,4 +129,6 @@ class SerializableSparkExecutionContext(val delegate: SparkExecutionContext) ext
   }
 
   override def getDataTracer(loggerName: String): DataTracer = delegate.getDataTracer(loggerName)
+
+  override def getTriggeringScheduleInfo: TriggeringScheduleInfo = delegate.getTriggeringScheduleInfo
```

Comment:
It is not common to have `null` value in Scala in general (which the `getTriggeringScheduleInfo` might return null).

One possible way to get around that is to have a separate interface to have the `getTriggeringScheduleInfo` method (e.g. `SchedulableProgramContext`), which program context, except the SparkExecutionContext, implements. The `SparkExecutionContext` can have it's own definition of `getTriggeringScheduleInfo`, which returns an `Option[TriggeringScheduleInfo]`.